base_hap.yaml

```diff
@@ -0,0 +1,103 @@
+---
+# HAP (HomeKit) enabled base configuration for RATGDO
+# This extends the base configuration with HomeKit Accessory Protocol support
+# Requires ESP32 (ESP8266 not supported by HAP-ESPHome)
+#
+# IMPORTANT: This configuration uses the "Production Ready" commit from HAP-ESPHome
+# (910e053b) that was marked as stable for garage door opener integration.
+# This commit predates the API compatibility issues introduced in later commits:
+# - Maintains proper hap_char_get_val() API usage
+# - Avoids HAP_VAL_TYPE_INVALID compatibility issues
+# - Includes proper header dependencies and type definitions
+
+external_components:
+  # Local RATGDO components
+  - source:
+      type: local
+      path: components
+    refresh: 1s
+    components: [ ratgdo ]
+  # HAP-ESPHome components for HomeKit support
+  # Uses "Production Ready" commit that maintains API compatibility
+  - source:
+      type: git
+      url: https://github.com/rednblkx/HAP-ESPHome
+      ref: 910e053b0ead74e73715b90ebf00e6a92ff66f18
+    refresh: 1s
+    components: [ homekit_base, homekit ]
+
+# Include base RATGDO configuration
+packages:
+  ratgdo_base:
+    url: https://github.com/ratgdo/esphome-ratgdo
+    files: [base.yaml]
+    refresh: 1s
+
+# HomeKit base component configuration
+# Default setup code: 159-35-728 (change for production use)
+# Setup ID: RGDO (4-character identifier for HomeKit)
+homekit_base:
+  meta:
+    name: "RATGDO HomeKit Bridge"
+    manufacturer: "RATGDO"
+    model: "ESPHome Garage Door Controller"
+    serial_number: "${id_prefix}"
+    fw_rev: "2.0-hap"
+  setup_code: '159-35-728'  # Consider changing for production deployment
+  setup_id: "RGDO"
```

Comment:
A hard coded setup ID can cause problems if two brand new devices are powered onto the same network and HomeKit tries to pair with one of them... the other could reply.  I solved this in my HomeKit firmware by generating a unique ID based on the lower 24-bits of MAC address.  The setup ID can be base-62 encoded [A-Z][a-z][0-9] which will encode 22-bits... good enough (I just drop the most significant character of the base-62 encoding.

With that you can then generate a QR code payload (text string X-HM://xxxxx) which you can send to the browser which can then generate a pairing QR code with a javascript library... so adding to Apple Home can be done by scanning the QR code.